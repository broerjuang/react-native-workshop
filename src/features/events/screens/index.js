//@flow

export {default as EventsScreen} from './EventsScreen';
