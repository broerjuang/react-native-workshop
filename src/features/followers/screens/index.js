// @flow
export {default as FollowersScreen} from './FollowersScreen.js';
