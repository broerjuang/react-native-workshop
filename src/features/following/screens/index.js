// @flow
export {default as FollowingScreen} from './FollowingScreen.js';
