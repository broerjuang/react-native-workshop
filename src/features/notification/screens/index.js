//@flow
export {default as AllNotificationsScreen} from './AllNotificationsScreen';
export {default as ParticipatingScreen} from './ParticipatingScreen';
export {default as UnreadScreen} from './UnreadScreen';
