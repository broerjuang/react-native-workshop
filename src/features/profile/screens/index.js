export {default as ProfileScreen} from './ProfileScreen';
export {default as SettingScreen} from './SettingScreen';
