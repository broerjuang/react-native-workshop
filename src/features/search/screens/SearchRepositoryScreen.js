// @flow
import React from 'react';
import {ScrollView} from 'react-native';
import {RepoCard} from '../../../global/core-ui/index';
import {connect} from 'react-redux';

import type {NavigationScreenProp} from 'react-navigation';

type Repo = {
  full_name: string;
  description: string;
  stargazers_count: number;
  forks_count: number;
  language: string;
  fork: boolean;
};

type Props = {
  navigation: NavigationScreenProp<[]>;
  repos: Array<Repo>;
};

function SearchRepositoryScreen(props: Props) {
  let {repos, navigation} = props;
  return (
    <ScrollView style={{flex: 1, backgroundColor: '#fff'}}>
      {repos
        ? repos.map((repo, index) => {
          return (
            <RepoCard
              key={index}
              repo={repo}
              onPress={() => {
                navigation.navigate('RepositoryDetailScreen');
              }}
            />
          );
        })
        : null}
    </ScrollView>
  );
}

function mapStateToProps(state) {
  return {
    repos: state.searchReducer.repos,
  };
}

export default connect(mapStateToProps)(SearchRepositoryScreen);
