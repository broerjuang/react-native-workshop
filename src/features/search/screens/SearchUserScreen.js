// @flow

import React, {Component} from 'react';
import {
  Text,
  Image,
  TouchableOpacity,
  ScrollView,
  StyleSheet,
} from 'react-native';
import {RowWith3Column, Icon} from '../../../global/core-ui/index';
import {connect} from 'react-redux';

import type {NavigationScreenProp} from 'react-navigation';

type User = {
  login: string,
  avatar_url: string,
};

type Object = {
  navigation: NavigationScreenProp<[]>,
  users: Array<User>,
};

class SearchUserScreen extends Component<Object> {
  render() {
    const {list} = styles;
    let {users} = this.props;
    return (
      <ScrollView style={{flex: 1, backgroundColor: '#fff'}}>
        {users.map((user, index) => {
          return (
            <TouchableOpacity
              key={index}
              style={list}
              onPress={() => this._openUser(user)}
            >
              <RowWith3Column
                left={
                  <Image
                    style={{width: 32, height: 32, borderRadius: 16}}
                    source={{uri: user.avatar_url}}
                  />
                }
                content={<Text>{user.login}</Text>}
                right={
                  <Icon
                    name="angle-right"
                    size={24}
                    color="grey"
                    type="FONTAWESOME"
                  />
                }
              />
            </TouchableOpacity>
          );
        })}
      </ScrollView>
    );
  }

  _openUser(user: User) {
    this.props.navigation.navigate('ProfileScreen');
  }
}

const styles = StyleSheet.create({
  list: {
    borderTopWidth: 1,
    borderColor: '#d6d4d4',
    paddingVertical: 6,
  },
});

function mapStateToProps(state) {
  return {
    users: state.searchReducer.users,
  };
}

export default connect(mapStateToProps)(SearchUserScreen);
