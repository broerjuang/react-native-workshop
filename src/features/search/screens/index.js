export {default as SearchRepositoryScreen} from './SearchRepositoryScreen';
export {default as SearchUserScreen} from './SearchUserScreen';
