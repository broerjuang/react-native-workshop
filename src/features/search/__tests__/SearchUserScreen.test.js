import renderer from 'react-test-renderer';
import React from 'react';
import {SearchUserScreen} from '../screens/SearchUserScreen';
describe('container test', () => {
  it('should render LoginScreen corectly', () => {
    let component = renderer.create(
      <SearchUserScreen
        users={['aji', 'deany', 'ari', 'jo', 'astrid']}
        navigation={{navigate: (s) => s}}
        handleAction={(action) => {
          return;
        }}
        token={'sd'}
        isLogin={false}
        userName={''}
        onRequest={false}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
