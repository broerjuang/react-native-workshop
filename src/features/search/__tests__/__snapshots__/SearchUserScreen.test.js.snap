// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container test should render LoginScreen corectly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#d6d4d4",
          "borderTopWidth": 1,
          "opacity": 1,
          "paddingVertical": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "15%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 16,
                  "height": 32,
                  "width": 32,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexWrap": "wrap",
                "justifyContent": "center",
                "padding": 10,
                "width": "70%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "15%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#d6d4d4",
          "borderTopWidth": 1,
          "opacity": 1,
          "paddingVertical": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "15%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 16,
                  "height": 32,
                  "width": 32,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexWrap": "wrap",
                "justifyContent": "center",
                "padding": 10,
                "width": "70%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "15%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#d6d4d4",
          "borderTopWidth": 1,
          "opacity": 1,
          "paddingVertical": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "15%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 16,
                  "height": 32,
                  "width": 32,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexWrap": "wrap",
                "justifyContent": "center",
                "padding": 10,
                "width": "70%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "15%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#d6d4d4",
          "borderTopWidth": 1,
          "opacity": 1,
          "paddingVertical": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "15%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 16,
                  "height": 32,
                  "width": 32,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexWrap": "wrap",
                "justifyContent": "center",
                "padding": 10,
                "width": "70%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "15%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#d6d4d4",
          "borderTopWidth": 1,
          "opacity": 1,
          "paddingVertical": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "15%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 16,
                  "height": 32,
                  "width": 32,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexWrap": "wrap",
                "justifyContent": "center",
                "padding": 10,
                "width": "70%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "15%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
